koku/masu/database/report_db_accessor_base.py

```diff
@@ -153,6 +164,27 @@ def _execute_trino_raw_sql_query_with_description(
         LOG.info(log_json(msg="executed trino sql", log_ref=log_ref, running_time=running_time, context=ctx))
         return results, description
 
+    def _handle_trino_external_error(
+        self, ex, sql, sql_params, context, log_ref, attempts_left, trino_external_error_retries, conn_params, ctx
```

Comment:
An interface with this many parameters should require the use of kwargs. It's too easy to get parameters out of order causing confusing errors.

Reply:
To add to this, `context` and `ctx` should represent the same thing, so we really shouldn't be passing both of these to a function. Admittedly, this `extract_context_from_sql_params` is not implemented very well and is probably overly complicated, but still, we should just pick one or the other to log instead of both.